### PR TITLE
Update Python test matrix (drop EOL Pythons and add new ones)'

### PR DESCRIPTION
**Pull request by jdufresne**
Python 2.6 and 3.3 are EOL. They are no longer receiving bug fixes,
including for security issues. Python 2.6 went EOL on 2013-10-29 and
Python 3.3 went EOL on 2017-09-29. For details, see:

https://devguide.python.org/#status-of-python-branches

Additionally, the latest pytest release dropped support for these Python
versions. This change is causing CI failures for Python 2.6 and 3.3. For
details on the pytest changes, see:

https://pytest.readthedocs.io/en/latest/changelog.html#pytest-3-3-0-2017-11-23

For additional discussion on why project maintainers may wish to drop
Python 2.6 support, please see:

https://snarky.ca/stop-using-python-2-6/
http://www.curiousefficiency.org/posts/2015/04/stop-supporting-python26.html

Reduces testing resources.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -9,7 +9,7 @@ This is a Python implementation of the `WHATWG Encoding standard
   https://github.com/gsnedders/python-webencodings
 * PyPI releases: http://pypi.python.org/pypi/webencodings
 * License: BSD
-* Python 2.6+ and 3.3+
+* Python 2.7+ and 3.4+
```
Comment (hugovk) [inline README.rst]: "Python 2.7 and 3.4+"

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
Here's the pip installs for webencodings from PyPI for last month, showing very low 2.6 and 3.3 numbers:

| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 2.7            |   42.1% |        204,177 |
| 3.6            |   22.8% |        110,262 |
| 3.5            |   20.0% |         96,819 |
| 3.4            |   14.8% |         71,670 |
| 3.7            |    0.2% |            953 |
| 3.3            |    0.1% |            361 |
| 2.6            |    0.1% |            253 |
| None           |    0.0% |              4 |
| 2.8            |    0.0% |              1 |
| 3.2            |    0.0% |              1 |

Source: `pypinfo --start-date -47 --end-date -17 --percent --pip --markdown webencodings pyversion`

PS Looks like this job https://travis-ci.org/gsnedders/python-webencodings/jobs/325678092 failed due to a temporary network error. Restarting it will probably sort it.

**Comment by jdufresne** [comment]
Thanks for the review. I've modified the README as suggested. Tests are now green.

**Comment by jdufresne** [comment]
Since this PR originally went up, there have been new Python releases and now Python 3.4 is EOL. I have updated the test matrix to test all new Pythons.

**Review by hugovk — APPROVED** [review]

**Review by Giangpro89 — APPROVED** [review]

**Pull request closed without merging**